vendor/github.com/coreos/go-oidc/oidc/provider.go

```diff
@@ -353,9 +353,6 @@ func (p ProviderConfig) Valid() error {
 	if !contains(p.IDTokenSigningAlgValues, "RS256") {
 		return errors.New("id_token_signing_alg_values_supported must include 'RS256'")
 	}
-	if contains(p.TokenEndpointAuthMethodsSupported, "none") {
```

Comment:
We don't use these values in go-oidc, kubectl or the API server. We can add the check, but Kubernetes isn't impacted in any way.

We always assume that the ID token is signed using `RS256`, because providers have to default to that value and there's no standard way of overriding it. That default has saved us in the past from bad providers (e.g. when one returned `HS256`), and no one's complained that `RS256` didn't work for them.